### PR TITLE
Utilizing new dispatcher objects within the gateway

### PR DESCRIPTION
**Pull request by Jason-RH**
First go at breaking down the client's responsibilities. ResponseDispatcher handles messages from the reader and dispatches them (in this case putting them on the kafka queue) if necessary.

@dehort Let me know what you think on the approach!

### CODE REVIEW

**Comment by Jason-RH** [comment]
I also included some initial pieces for the WorkDispatcher, but that'll most likely be completed in a separate PR

File: cmd/gateway/main.go [hunk 1]
```diff
@@ -25,7 +25,8 @@ func main() {
 	wsMux := mux.NewRouter()
 	cm := c.NewConnectionManager()
 	kw := queue.StartProducer(queue.GetProducer())
-	rc := ws.NewReceptorController(cm, wsMux, kw)
+	d := c.NewResponseDispatcherFactory(kw)
+	rc := ws.NewReceptorController(cm, wsMux, kw, d)
```
Comment (dehort) [inline cmd/gateway/main.go]: Can we avoid passing the KafkaWriter instance (kw) to the ReceptorController constructor now?

Comment (Jason-RH) [reply cmd/gateway/main.go]: Good catch yeah we can avoid that. Will push an update

**Comment by Jason-RH** [comment]
@dehort Okay had some rebasing issues so sry for the extra commits. Everything looks good now on my end though, and yes the work dispatcher code is still in progress and will be updated in a different PR once that's complete.

**Pull request merged**